### ISSUE TITLE
Depthwise F.conv1d produces wrong weight.grad on XPU when input spatial size is 1 with a non-packed stride

### ISSUE THREAD

**Issue by kaixuanliu** [intel/torch-xpu-ops#4906]
### 🐛 Describe the bug

On XPU, the weight gradient of a depthwise (groups == in_channels) F.conv1d is silently wrong when the input has a size-1 spatial dimension whose stride is not 1. x.grad is correct; only weight.grad is affected, so it fails silently during training.

Such tensors arise naturally from rearrange(x, "b s d -> b d s") / .transpose(1, 2) on a single-token input — the common decode-step layout for Mamba/SSM-style models. 
```
import torch, torch.nn.functional as F
torch.manual_seed(0)
batch, dim, width = 2, 4, 2

def grads(device, x, weight, grad_out):
    x = x.to(device).detach().requires_grad_()
    weight = weight.to(device).detach().requires_grad_()
    F.conv1d(x, weight.unsqueeze(1), None, padding=width-1, groups=dim)[..., :x.shape[-1]].backward(grad_out.to(device))
    return weight.grad.cpu()

x = torch.randn(batch, 1, dim).transpose(1, 2)   # (2, 4, 1), stride (4, 1, 4)
w, g = torch.randn(dim, width), torch.randn(batch, dim, 1)
print("cpu:", grads("cpu", x, w, g).flatten().tolist())
print("xpu:", grads("xpu", x, w, g).flatten().tolist())
```
it returns:
```
cpu: [0.0, -1.274393081665039, 0.0, -2.146540403366089, 0.0, 2.3225393295288086, 0.0, -0.20848023891448975]
xpu: [-2.393806219100952, -1.2743932008743286, 0.0, 0.0, 3.6005606651306152, -0.22546251118183136, 0.0, 0.0]
```

### Versions

```
PyTorch version: 2.13.0+xpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.4 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04.1) 13.3.0
Clang version: Could not collect
CMake version: version 3.28.3
Libc version: glibc-2.39

Python version: 3.12.3 (main, Jun 19 2026, 12:46:00) [GCC 13.3.0] (64-bit runtime)
Python platform: Linux-6.17.0-5-generic-x86_64-with-glibc2.39
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: True
XPU used to build PyTorch: 20260000
Intel GPU driver version:
* intel-opencl-icd:     26.22.38646.6-1~24.04~ppa1
* libze1:       1.28.6-1~24.04~ppa1
Intel GPU models onboard:
N/A
Intel GPU models detected:
* [0] _XpuDeviceProperties(name='Intel(R) Arc(TM) Pro B60 Graphics', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero V2', type='gpu', device_id=0xE211, uuid=868011e2-0000-0000-1700-000000000000, driver_version='1.15.38646+6', total_memory=24480MB, local_mem_size=128KB, last_level_cache_size=18432KB, max_compute_units=160, memory_clock_rate=2400MHz, memory_bus_width=64-bit, gpu_eu_count=160, gpu_subslice_count=20, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1, is_integrated_gpu=0)
* [1] _XpuDeviceProperties(name='Intel(R) Arc(TM) Pro B60 Graphics', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero V2', type='gpu', device_id=0xE211, uuid=868011e2-0000-0000-2c00-000000000000, driver_version='1.15.38646+6', total_memory=24480MB, local_mem_size=128KB, last_level_cache_size=18432KB, max_compute_units=160, memory_clock_rate=2400MHz, memory_bus_width=64-bit, gpu_eu_count=160, gpu_subslice_count=20, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1, is_integrated_gpu=0)
* [2] _XpuDeviceProperties(name='Intel(R) Arc(TM) Pro B60 Graphics', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero V2', type='gpu', device_id=0xE211, uuid=868011e2-0000-0000-3d00-000000000000, driver_version='1.15.38646+6', total_memory=24480MB, local_mem_size=128KB, last_level_cache_size=18432KB, max_compute_units=160, memory_clock_rate=2400MHz, memory_bus_width=64-bit, gpu_eu_count=160, gpu_subslice_count=20, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1, is_integrated_gpu=0)
* [3] _XpuDeviceProperties(name='Intel(R) Arc(TM) Pro B60 Graphics', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero V2', type='gpu', device_id=0xE211, uuid=868011e2-0000-0000-4e00-000000000000, driver_version='1.15.38646+6', total_memory=24480MB, local_mem_size=128KB, last_level_cache_size=18432KB, max_compute_units=160, memory_clock_rate=2400MHz, memory_bus_width=64-bit, gpu_eu_count=160, gpu_subslice_count=20, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1, is_integrated_gpu=0)
* [4] _XpuDeviceProperties(name='Intel(R) Arc(TM) Pro B60 Graphics', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero V2', type='gpu', device_id=0xE211, uuid=868011e2-0000-0000-9700-000000000000, driver_version='1.15.38646+6', total_memory=24480MB, local_mem_size=128KB, last_level_cache_size=18432KB, max_compute_units=160, memory_clock_rate=2400MHz, memory_bus_width=64-bit, gpu_eu_count=160, gpu_subslice_count=20, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1, is_integrated_gpu=0)
* [5] _XpuDeviceProperties(name='Intel(R) Arc(TM) Pro B60 Graphics', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero V2', type='gpu', device_id=0xE211, uuid=868011e2-0000-0000-a900-000000000000, driver_version='1.15.38646+6', total_memory=24480MB, local_mem_size=128KB, last_level_cache_size=18432KB, max_compute_units=160, memory_clock_rate=2400MHz, memory_bus_width=64-bit, gpu_eu_count=160, gpu_subslice_count=20, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1, is_integrated_gpu=0)
* [6] _XpuDeviceProperties(name='Intel(R) Arc(TM) Pro B60 Graphics', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero V2', type='gpu', device_id=0xE211, uuid=868011e2-0000-0000-ba00-000000000000, driver_version='1.15.38646+6', total_memory=24480MB, local_mem_size=128KB, last_level_cache_size=18432KB, max_compute_units=160, memory_clock_rate=2400MHz, memory_bus_width=64-bit, gpu_eu_count=160, gpu_subslice_count=20, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1, is_integrated_gpu=0)
* [7] _XpuDeviceProperties(name='Intel(R) Arc(TM) Pro B60 Graphics', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero V2', type='gpu', device_id=0xE211, uuid=868011e2-0000-0000-cb00-000000000000, driver_version='1.15.38646+6', total_memory=24480MB, local_mem_size=128KB, last_level_cache_size=18432KB, max_compute_units=160, memory_clock_rate=2400MHz, memory_bus_width=64-bit, gpu_eu_count=160, gpu_subslice_count=20, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1, is_integrated_gpu=0)
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: False
Caching allocator config: N/A

CPU:
Architecture:                            x86_64
CPU op-mode(s):                          32-bit, 64-bit
Address sizes:                           52 bits physical, 57 bits virtual
Byte Order:                              Little Endian
CPU(s):                                  512
On-line CPU(s) list:                     0-511
Vendor ID:                               GenuineIntel
BIOS Vendor ID:                          Intel(R) Corporation
Model name:                              Intel(R) Xeon(R) 6980P
BIOS Model name:                         Intel(R) Xeon(R) 6980P  CPU @ 2.0GHz
BIOS CPU family:                         179
CPU family:                              6
Model:                                   173
Thread(s) per core:                      2
Core(s) per socket:                      128
Socket(s):                               2
Stepping:                                1
CPU(s) scaling MHz:                      21%
CPU max MHz:                             3900.0000
CPU min MHz:                             800.0000
BogoMIPS:                                4000.00
Flags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cat_l2 cdp_l3 intel_ppin cdp_l2 ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect user_shstk avx_vnni avx512_bf16 wbnoinvd dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req vnmi avx512vbmi umip pku ospke waitpkg avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg avx512_vpopcntdq la57 rdpid bus_lock_detect cldemote movdiri movdir64b enqcmd fsrm md_clear serialize tsxldtrk pconfig arch_lbr ibt amx_bf16 avx512_fp16 amx_tile amx_int8 flush_l1d arch_capabilities
Virtualization:                          VT-x
L1d cache:                               12 MiB (256 instances)
L1i cache:                               16 MiB (256 instances)
L2 cache:                                512 MiB (256 instances)
L3 cache:                                1008 MiB (2 instances)
NUMA node(s):                            2
NUMA node0 CPU(s):                       0-127,256-383
NUMA node1 CPU(s):                       128-255,384-511
Vulnerability Gather data sampling:      Not affected
Vulnerability Ghostwrite:                Not affected
Vulnerability Indirect target selection: Not affected
Vulnerability Itlb multihit:             Not affected
Vulnerability L1tf:                      Not affected
Vulnerability Mds:                       Not affected
Vulnerability Meltdown:                  Not affected
Vulnerability Mmio stale data:           Not affected
Vulnerability Old microcode:             Not affected
Vulnerability Reg file data sampling:    Not affected
Vulnerability Retbleed:                  Not affected
Vulnerability Spec rstack overflow:      Not affected
Vulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:                Mitigation; Enhanced / Automatic IBRS; IBPB conditional; PBRSB-eIBRS Not affected; BHI BHI_DIS_S
Vulnerability Srbds:                     Not affected
Vulnerability Tsa:                       Not affected
Vulnerability Tsx async abort:           Not affected
Vulnerability Vmscape:                   Mitigation; IBPB before exit to userspace

Versions of relevant libraries:
[pip3] apollo-torch==1.0.3
[pip3] dpcpp-cpp-rt==2026.0.0
[pip3] galore-torch==1.0
[pip3] impi-rt==2021.18.0
[pip3] intel-cmplr-lib-rt==2026.0.0
[pip3] intel-cmplr-lib-ur==2026.0.0
[pip3] intel-cmplr-lic-rt==2026.0.0
[pip3] intel-opencl-rt==2026.0.0
[pip3] intel-openmp==2026.0.0
[pip3] intel-pti==0.17.0
[pip3] intel-sycl-rt==2026.0.0
[pip3] mkl==2026.0.0
[pip3] mypy_extensions==1.1.0
[pip3] numpy==2.2.6
[pip3] oneccl==2022.0.0
[pip3] oneccl-devel==2022.0.0
[pip3] onemkl-license==2026.0.0
[pip3] onemkl-sycl-blas==2026.0.0
[pip3] onemkl-sycl-dft==2026.0.0
[pip3] onemkl-sycl-lapack==2026.0.0
[pip3] onemkl-sycl-rng==2026.0.0
[pip3] onemkl-sycl-sparse==2026.0.0
[pip3] onnx==1.22.0
[pip3] pytorch-msssim==1.0.0
[pip3] tbb==2023.0.0
[pip3] tcmlib==1.5.0
[pip3] torch==2.13.0+xpu
[pip3] torchao==0.17.0
[pip3] torchaudio==2.11.0+xpu
[pip3] torchcodec==0.14.0+cpu
[pip3] torchdata==0.11.0
[pip3] torchvision==0.28.0+xpu
[pip3] triton==3.7.1
[pip3] triton-xpu==3.7.2
[pip3] umf==1.1.0
```

**Comment by kaixuanliu** [comment]
@LuFinch Hi, can you help take a look at this?

**Comment by LuFinch** [comment]
sure, it is WIP.

**Comment by LuFinch** [comment]
@kaixuanliu Fix PR merged. Please verify with today nightly whl.

**Comment by kaixuanliu** [comment]
I checked w/ PT 2.15.0.dev20260824+xpu, the issue still exists.

**Comment by LuFinch** [comment]
Seems 0824 does't include this commit yet. Lets try 0825

**Comment by kaixuanliu** [comment]
Verified w/ PT 2.15.0.dev20260825+xpu, thx! close the issue.
